File: src/lib/api/hooks.ts
import { invoke } from "@tauri-apps/api/core";
import type { HooksStatus } from "@/types";

export const hooksApi = {
  async getStatus(): Promise<HooksStatus> {
    return await invoke("get_hooks_status");
  },
  async install(tool: string): Promise<void> {
    return await invoke("install_hook", { tool });
  },
  async uninstall(tool: string): Promise<void> {
    return await invoke("uninstall_hook", { tool });
  },
};

File: src-tauri/src/commands/hooks.rs
use crate::models::HooksStatus;

#[tauri::command]
pub fn get_hooks_status() -> Result<HooksStatus, String> {
    crate::hooks::get_hooks_status().map_err(|e| e.to_string())
}

#[tauri::command]
pub fn install_hook(tool: String) -> Result<(), String> {
    crate::hooks::install_hooks(&tool).map_err(|e| e.to_string())
}

#[tauri::command]
pub fn uninstall_hook(tool: String) -> Result<(), String> {
    crate::hooks::uninstall_hooks(&tool).map_err(|e| e.to_string())
}
